src/v2/Artsy/Relay/middleware/cache/Cache.ts

```diff
@@ -0,0 +1,102 @@
+import { QueryResponseCache } from "relay-runtime"
+import createLogger from "v2/Utils/logger"
+import RelayQueryResponseCache from "relay-runtime/lib/network/RelayQueryResponseCache"
+import { isServer } from "lib/environment"
+
+const logger = createLogger("v2/Artsy/middleware/cache/Cache")
+
+export interface CacheConfig {
+  size: number
+  ttl: number
+}
+
+export class Cache {
+  cacheConfig: CacheConfig
+  relayCache: RelayQueryResponseCache
+
+  redisCache: {
+    del: (key: string) => Promise<void>
+    expire: (key: string, ttl: number) => Promise<void>
+    flushall: () => Promise<void>
+    get: (key: string) => Promise<string>
+    set: (key: string, value: string) => Promise<void>
+  }
+
+  constructor(cacheConfig: CacheConfig) {
+    this.cacheConfig = cacheConfig
+    this.initRelayCache()
+    this.initRedisCache()
+  }
+
+  initRelayCache() {
+    this.relayCache = new QueryResponseCache(this.cacheConfig)
+  }
+
+  initRedisCache() {
+    if (!isServer) {
+      return
+    }
+
+    const { promisify } = require("util")
+    const redis = require("redis")
+    const client = redis.createClient()
+
+    this.redisCache = {
+      del: promisify(client.del).bind(client),
+      expire: promisify(client.expire).bind(client),
+      flushall: promisify(client.flushall).bind(client),
+      get: promisify(client.get).bind(client),
+      set: promisify(client.set).bind(client),
+    }
+  }
+
+  getCacheKey(query, variables) {
+    const cacheKey = JSON.stringify({ query, variables })
```

Comment:
This will generate very large cache keys, impacting the latency of communicating with the cache and adding unnecessarily to storage. I'd probably hash these to generate a shorter, consistent, unique string.

Reply:
@joeyAghion - this isn't storying the query itself, but rather the query name: 
<img width="709" alt="Screen Shot 2020-06-02 at 1 54 17 PM" src="https://user-images.githubusercontent.com/236943/83568901-93ee9880-a4d8-11ea-9310-ee7b5830221c.png">

Reply:
OH 🤦

Reply:
I kept the naming consistent across the other bit of middleware i adapted but will update the variable name for clairity, its understandably confusing 👍